### DISCUSSION TITLE
Object browser: show only filtered members

### DISCUSSION THREAD

**Discussion by garetjax76** [codefori/vscode-ibmi#2220] [Ideas]
First of all thanks for this incredibile extension.
As a current user of Rational Developer for I, I'm starting to use it and it's really impressive, but there's only one thing that keeps me using RDI: the member filters.
Let me explain.
In my company, legacy sources are scattered in two or three libraries and many files (each branch has their own files, divided per source types). When I have to work on a ticket or a project, I often have to look into multiple libraries and files searching for members that have the same prefix.
In RDI, there's a very handy feature called "member filter" - similar to object browser, that actively looks into the specified source files (that could also be "*"), but unlike the object browser, it shows ONLY the found members.
I looked in the settings and in the documentation but seems to me there is not option to do this in codefori.

For example, I can create a filter like this:
![image](https://github.com/user-attachments/assets/86fe52ac-e72c-4668-a42b-8dd13f935b4c)
The result is this one (the sources are in 3 diffferent files):
![image](https://github.com/user-attachments/assets/c9409226-650b-4c6c-ab52-45729e183a29)
It's very convienent because it searches through all the source files, and for quite large projects there could be many source files involved.

In CodeForI object browser, I create a filter with * files and CFFREG* members, it obviously lists ALL the sources in the library, then I have to manually open each file to find out if there are some related members (and, honestly, it's ugly and inconvient).

I understand that this situation, in my case, could be solved by a source reordering, but it would be a very time consuming task - not in budget right now and maybe will never be. 
This is just for pointing out something that RDi does better and would be very useful also in this extension.

Again thanks for your work.

**Comment by rustygad** [comment]
I'm also an RDi user and find it's filter is better presenting a list of of matching members without being embedded within a source library. In Code4i it's a hassle having to expand the source files to see the filtered members.
